driver.go

```diff
@@ -105,21 +106,12 @@ func (d SnowflakeDriver) Open(dsn string) (driver.Conn, error) {
 	}
 
 	sc.populateSessionParameters(authData.Parameters)
+	sc.ctx = context.WithValue(sc.ctx, SFSessionIDKey, authData.SessionID)
```

Comment:
This is a good idea. Aside from this, perhaps it will be good to have the query id too so that each log line can have the query id instead to match queries and logs. The same idea can apply to all other drivers.

Reply:
Note that enhancement is not necessarily be in this PR, though.

Reply:
The query Id was obtained too late only after we got response from a request. For now I would think we could correlate query ID with session ID to find the query related logs. Or We could add a local query ID when we start a query. These IDs I think we could sync with the request ID concept numeracy is introducing and we could easily add late in QueryWithContext and ExecWithContext.